File: tests/input_files/IOTestsComparison/TestAlohaWriter/short_F77writer_complex_mass_scheme/sss1_1.f
subroutine SSS1_1(S2, S3, COUP, M1,S1)
use aloha_object
use model_object
implicit none
 complex*16 CI
 parameter (CI=(0d0,1d0))
 complex*16 COUP
 complex*16 M1
 real*8 P1(0:3)
 type(aloha) S1
 type(aloha) S2
 type(aloha) S3
 complex*16 denom
entry SSS1_2(S2, S3, COUP, M1,S1)

entry SSS1_3(S2, S3, COUP, M1,S1)

    S1%%P(:) = +S2%%P(:)+S3%%P(:)
P1(:) = -S1 %% P (:)
    denom = COUP/(P1(0)**2-P1(1)**2-P1(2)**2-P1(3)**2 - M1**2)
    S1%%W(1)= denom*CI * S3 %% W(1)*S2 %% W(1)
 end

File: tests/input_files/IOTestsComparison/TestAlohaWriter/short_Cwriter_complex_mass_scheme/sss1_1.c
#include "SSS1_1.h"

 void SSS1_1(ALOHAOBJ S2, ALOHAOBJ S3, std::complex<double> COUP, std::complex<double> M1,ALOHAOBJ  & S1)
{
static std::complex<double> cI = std::complex<double>(0.,1.);
 double  P1[4];
 std::complex<double>  denom;
    S1.p[0] = +S2.p[0]+S3.p[0];
    S1.p[1] = +S2.p[1]+S3.p[1];
    S1.p[2] = +S2.p[2]+S3.p[2];
    S1.p[3] = +S2.p[3]+S3.p[3];
P1[0] = -S1.p[0];
P1[1] = -S1.p[1];
P1[2] = -S1.p[2];
P1[3] = -S1.p[3];
    denom = COUP/((P1[0]*P1[0])-(P1[1]*P1[1])-(P1[2]*P1[2])-(P1[3]*P1[3]) - (M1*M1));
    S1.W[0]= denom*cI * S3.W[0]*S2.W[0];
}

 void SSS1_2(ALOHAOBJ S2, ALOHAOBJ S3, std::complex<double> COUP, std::complex<double> M1,ALOHAOBJ  & S1)
{

 SSS1_1(S2,S3,COUP,M1,S1);
}
 void SSS1_3(ALOHAOBJ S2, ALOHAOBJ S3, std::complex<double> COUP, std::complex<double> M1,ALOHAOBJ  & S1)
{

 SSS1_1(S2,S3,COUP,M1,S1);
}
